src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.contribution.ts

```diff
@@ -339,5 +338,6 @@ configurationRegistry.registerConfiguration({
 	}
 });
 
-Registry.as<IWorkbenchContributionsRegistry>(WorkbenchExtensions.Workbench)
-	.registerWorkbenchContribution(StartupPageContribution, LifecyclePhase.Restored);
+registerWorkbenchContribution2(WorkspacePlatformContribution.ID, WorkspacePlatformContribution, WorkbenchPhase.AfterRestored);
+registerWorkbenchContribution2(StartupPageEditorResolverContribution.ID, StartupPageEditorResolverContribution, WorkbenchPhase.BlockRestore);
+registerWorkbenchContribution2(StartupPageRunnerContribution.ID, StartupPageRunnerContribution, WorkbenchPhase.AfterRestored);
```

Comment:
Can we leave it to run based on `editorId`, that should work, no?

Reply:
No. If no welcome page is opened before restore (example via OpenWalkthrough command), we need the Restored phase to trigger the StartupPageRunnerContribution to be instantiated so that the `workbench.startupEditor` can be setup correctly.

Reply:
But are we still fixing https://github.com/microsoft/vscode/issues/193623 then?

Reply:
Oh I see, its fixed due to the resolver.